internal/services/firewall/firewall_resource.go

```diff
@@ -159,15 +160,43 @@ func resourceFirewall() *pluginsdk.Resource {
 			},
 
 			"dns_servers": {
-				Type:     pluginsdk.TypeList,
-				Optional: true,
-				MinItems: 1,
+				Type:       pluginsdk.TypeList,
+				Optional:   true,
+				Computed:   true,
+				MinItems:   1,
+				Deprecated: "Use the separate `dns_settings` for more DNS settings",
 				Elem: &pluginsdk.Schema{
 					Type:         pluginsdk.TypeString,
 					ValidateFunc: validation.IsIPAddress,
 				},
 			},
 
+			// dns_proxy_enabled is NOT related to dns_servers in fact, they are independent
+			// see: https://github.com/hashicorp/terraform-provider-azurerm/pull/8878#discussion_r527355050
+			"dns_settings": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*schema.Schema{
+						"custom_servers": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.IsIPAddress,
+							},
+						},
+
+						"proxy_enabled": {
```

Comment:
Thanks for your suggestion. i have tried to move it to the top level, but then it is not compatiable with what Portal does. for portal there is a separate switch to turn off/on the DNS configurations:

**Before Turn On**
![image](https://user-images.githubusercontent.com/2633022/229431462-e1c80000-ab6c-43db-a9b4-a9b0eec21bf3.png)

**After Turn On**
![image](https://user-images.githubusercontent.com/2633022/229431319-46ebbd8e-c7bc-410c-b6ef-36913f20a24b.png)

Reply:
i don't understand why its not compatible ? you can leave the existing servers property and then add a `dns_proxy_enabled` property instead of creating a new block?

Reply:
with a top-level `dns_proxy_enabled` property, no matter the value is `true` or `false`, the DNS settings will be `Enabled`. then how can we set the DNS Setting to `Disabled`?

Reply:
you can add a default of false?

Reply:
we are already passing in the custom servers?

i don't follow how it will be a problem to have it outside the block vs in? the radio option "enabling disabling" it in the portal appears to be purely cosmetic

Reply:
here we set the `dns_proxy_enabled` as top level:

```go
	"dns_servers": {
		Type:     pluginsdk.TypeList,
		Optional: true,
		MinItems: 1,
		Elem: &pluginsdk.Schema{
			Type:         pluginsdk.TypeString,
			ValidateFunc: validation.IsIPAddress,
		},
	},

	"dns_proxy_enabled": {
		Type:     pluginsdk.TypeBool,
		Optional: true,
		Computed: true,
	},
```

then with tf configuration like below will also Enable the DNS Settings and set the `DNS Proxy` to `Disabled`. it may action just the same as not enable DNS settings. but i guess it makes a diff between terraform-provider and portal and it could lead to issues down the line.

```tf
resource "azurerm_firewall" "test" {
  name                = "acctestfirewall"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku_name            = "AZFW_VNet"
  sku_tier            = "Standard"
}
```

i have actually implemented the desing of make `dns_proxy_enabled` as a top-level and have tested that it works for current version. i can push it if you are ok with it.

Reply:
What "enables dns settings" in the api? from the screenshot all it seems to do is allow you to configure those two options?

Reply:
there is no specific field in the API to enable DNS Settings, but just set fields to a map as field `AdditionalProperties` as below:

```go
AdditionalProperties["Network.DNS.EnableProxy"] = "true"
AdditionalProperties["Network.DNS.Servers"] = "1.1.1.1,2.2.2.2"
```
from the screenshot below and API requests in the Portal, when `Disabled` selected, we should not set the `AdditionalProperties` fields. but as previous discusstion as top-level `dns_proxy_enabled` property, we'll always set the `AdditionalProperties["Network.DNS.EnableProxy"] ` field.

![image](https://user-images.githubusercontent.com/2633022/229690906-c9412134-9e18-4218-b4f2-5e063caee7b7.png)

code of how `AdditionalProperties` is set:
https://github.com/hashicorp/terraform-provider-azurerm/pull/20519/files#diff-9b29e7d553e1b06cad14665d2ee767d5ffeefad32821d783e74f7372fbcde3d0R861-R865

Reply:
what you can do is only set it if the proxy is enabled

Reply:
in my test on Azure Portal, if only set `AdditionalProperties["Network.DNS.EnableProxy"] = "true"` it will just set the proxy is enabled (which means `DNS Servers` use the `Default (Azure provided)`) and this is what #9184 want.